cmd/server/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/argoproj/pkg/cli"
+	kubecli "github.com/argoproj/pkg/kube/cli"
+	"github.com/argoproj/pkg/stats"
+	"github.com/spf13/cobra"
+	"golang.org/x/net/context"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/argoproj/argo/cmd/server/apiserver"
+	wfclientset "github.com/argoproj/argo/pkg/client/clientset/versioned"
+	cmdutil "github.com/argoproj/argo/util/cmd"
+)
+
+const (
+	// CLIName is the name of the CLI
+	CLIName = "argo-server"
+)
+
+// NewRootCommand returns an new instance of the workflow-controller main entrypoint
+func NewRootCommand() *cobra.Command {
+	var (
+		clientConfig clientcmd.ClientConfig
+		logLevel     string // --loglevel
+		authType     string
+		configMap    string
+		port         int
+	)
+
+	var command = cobra.Command{
+		Use: CLIName,
+		RunE: func(c *cobra.Command, args []string) error {
+			cli.SetLogLevel(logLevel)
+			stats.RegisterStackDumper()
+			stats.StartStatsTicker(5 * time.Minute)
+
+			config, err := clientConfig.ClientConfig()
+			if err != nil {
+				return err
+			}
+			config.Burst = 30
+			config.QPS = 20.0
+
+			namespace, _, err := clientConfig.Namespace()
+			if err != nil {
+				return err
+			}
+
+			kubeConfig := kubernetes.NewForConfigOrDie(config)
+			wflientset := wfclientset.NewForConfigOrDie(config)
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			opts := apiserver.ArgoServerOpts{
+				Namespace:     namespace,
+				WfClientSet:   wflientset,
+				KubeClientset: kubeConfig,
+				RestConfig:    config,
+				AuthType:      authType,
+			}
+			err = opts.ValidateOpts()
+			if err != nil {
+				return err
+			}
+			apiServer := apiserver.NewArgoServer(opts)
+
+			go apiServer.Run(ctx, port)
+
+			// Wait forever
+			select {}
+
+		},
+	}
+
+	clientConfig = kubecli.AddKubectlFlagsToCmd(&command)
+	command.AddCommand(cmdutil.NewVersionCmd(CLIName))
+	command.Flags().IntVarP(&port, "port", "p", 2746, "Port to listen on")
+	command.Flags().StringVar(&authType, "auth-type", "server", "Enable authentication Type in Api Server. One of: client|server|hybrid")
```

Comment:
Flag should probably be `--auth-mode`.

Also there's no rhyme or reason in the choice of word casing in this sentence. The "API" Acronym is not capitalized and Type and Server are for some reason upper case.

Let's use this wording: "API server authentication mode. One of: client|server|hybrid"